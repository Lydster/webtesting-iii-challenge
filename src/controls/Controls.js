import React from "react";

const Controls = props => {
  const { locked, closed, toggleLocked, toggleClosed } = props;

  return (
    <div className="controls panel">
      <button
        title="lockButton"
        disabled={!closed}
        onClick={toggleLocked}
        className="toggle-btn"
      >
        {locked ? "Unlock Gate" : "Lock Gate"}
      </button>
      <button
        title="gateButton"
        disabled={locked}
        onClick={toggleClosed}
        className="toggle-btn"
      >
        {closed ? "Open Gate" : "Close Gate"}
      </button>
    </div>
  );
};

export default Controls;
