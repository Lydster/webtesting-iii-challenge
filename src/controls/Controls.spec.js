import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent } from "react-testing-library";
import "jest-dom/extend-expect";
import Controls from "./Controls";
import Display from "../display/Display";

describe("<Controls/>", () => {
  const display = render(<Display />);
  const control = render(<Controls />);
  it("matches control snapshot", () => {
    const controlTree = renderer.create(<Controls />);
    expect(controlTree.toJSON()).toMatchSnapshot();
  });
  it("check to see if text changes on click", () => {
    const { getByTitle } = render(<Controls />);
    const lockbutton = getByTitle(/lockButton/i);
    expect(lockbutton).toHaveTextContent(/Lock Gate/i);
    fireEvent.click(lockbutton);
    expect(lockbutton).toHaveTextContent(/Lock Gate/i);
  });

  it("if closed and locked cannot be opened", () => {
    const { getByTitle } = render(<Controls />);
    const closedButton = getByTitle(/gateButton/i);
    expect(closedButton).toHaveTextContent(/Close gate/i);
    fireEvent.click(closedButton);
    expect(closedButton).toHaveTextContent(/Close gate/i);
  });
});
