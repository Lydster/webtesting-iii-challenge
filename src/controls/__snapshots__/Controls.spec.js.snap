// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls/> matches control snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={true}
    title="lockButton"
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    title="gateButton"
  >
    Close Gate
  </button>
</div>
`;
